ios/RNSScreen.mm

```diff
@@ -1031,6 +1038,20 @@ - (void)viewDidLayoutSubviews
   }
 }
 
+- (BOOL)assertModalHierarchy
```

Comment:
Assertion should throw error / panic on failure, we just print the error.

```suggestion
- (BOOL)isModalWithHeader
```

I remember we talked about this method, sorry for misleading suggestion back then.

Reply:
My suggestion is to join two checks in this method:

```objective-c
return self.isModal && self.childViewControllers.count == 1 && [self.childViewControllers[0] isKindOfClass:UINavigationController.class]
```

Reply:
Resolved in https://github.com/software-mansion/react-native-screens/pull/1917/commits/add6ee3ea469736574a26cece43b2dbfcb714437.